app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/MixedUseController.scala

```diff
@@ -0,0 +1,429 @@
+    }
+
+  def updateAddress(
+    journey: EnteringSingleMixedUsePropertyDetails,
+    address: Address,
+    isManuallyEnteredAddress: Boolean
+  )(implicit hc: HeaderCarrier, request: Request[_]): EitherT[Future, Error, JourneyStatus] =
+    address match {
+      case _: NonUkAddress =>
+        EitherT.leftT[Future, JourneyStatus](
+          Error("Got non uk address in returns journey but expected uk address")
+        )
+      case a: UkAddress    =>
+        val answers            = journey.draftReturn.mixedUsePropertyDetailsAnswers.getOrElse(
+          IncompleteMixedUsePropertyDetailsAnswers.empty
+        )
+        val updatedDraftReturn = journey.draftReturn
+          .copy(mixedUsePropertyDetailsAnswers = Some(answers.unset(_.address).copy(address = Some(a))))
+
+        returnsService
+          .storeDraftReturn(
+            updatedDraftReturn,
+            journey.journey.subscribedDetails.cgtReference,
+            journey.journey.agentReferenceNumber
+          )
+          .map(_ => journey.journey.copy(draftReturn = updatedDraftReturn))
+
+    }
+
+  def isATrust(journey: EnteringSingleMixedUsePropertyDetails): Boolean =
+    journey.journey.subscribedDetails.isATrust
+
+  protected lazy val isUkCall: Call                                              = enterPostcodeCall
+  protected lazy val isUkSubmitCall: Call                                        = enterPostcodeCall
+  protected lazy val enterNonUkAddressCall: Call                                 = enterPostcodeCall
+  protected lazy val enterNonUkAddressSubmitCall: Call                           = enterPostcodeCall
+  protected lazy val backLinkCall: EnteringSingleMixedUsePropertyDetails => Call = _ => enterPostcodeCall
+  protected lazy val ukAddressNotAllowedExitPageCall: Option[Call]               = None
+
+  protected lazy val enterUkAddressCall: Call       = addressRoutes.MixedUseController.enterUkAddress()
+  protected lazy val enterUkAddressSubmitCall: Call = addressRoutes.MixedUseController.enterUkAddressSubmit()
+  protected lazy val enterPostcodeCall: Call        = addressRoutes.MixedUseController.enterPostcode()
+  protected lazy val enterPostcodeSubmitCall: Call  = addressRoutes.MixedUseController.enterPostcodeSubmit()
+  protected lazy val selectAddressCall: Call        = addressRoutes.MixedUseController.enterUkAddress()
+  protected lazy val selectAddressSubmitCall: Call  = addressRoutes.MixedUseController.selectAddress()
+  protected lazy val continueCall: Call             = addressRoutes.MixedUseController.checkYourAnswers()
+
+  def singleMixedUseGuidance(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val backLink =
+          r.draftReturn.mixedUsePropertyDetailsAnswers
+            .getOrElse(IncompleteMixedUsePropertyDetailsAnswers.empty)
```

Comment:
if you make a non-optional `IncompleteMixedUsePropertyDetailsAnswers` field available in `EnteringSingleMixedUsePropertyDetails` you'll only have to do `.getOrElse(IncompleteAnswers.empty)` in one place